src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Deployments/SaveAzureDeploymentTemplateCmdlet.cs

```diff
@@ -0,0 +1,92 @@
+    using Microsoft.WindowsAzure.Commands.Utilities.Common;
+
+    /// <summary>
+    /// Saves the deployment template to a file on disk.
+    /// </summary>
+    [Cmdlet(VerbsData.Save, "AzureRmDeploymentTemplate", SupportsShouldProcess = true,
+        DefaultParameterSetName = SaveAzureDeploymentTemplateCmdlet.DeploymentNameParameterSet), OutputType(typeof(PSObject))]
+    public class SaveAzureDeploymentTemplateCmdlet : ResourceManagerCmdletBase
+    {
+        /// <summary>
+        /// The deployment name parameter set.
+        /// </summary>
+        internal const string DeploymentNameParameterSet = "SaveByDeploymentName";
+
+        /// <summary>
+        /// The deployment object parameter set.
+        /// </summary>
+        internal const string DeploymentObjectParameterSet = "SaveByDeploymentObject";
+
+        /// <summary>
+        /// Gets or sets the deployment name parameter.
+        /// </summary>
+        [Alias("Name")]
+        [Parameter(ParameterSetName = SaveAzureDeploymentTemplateCmdlet.DeploymentNameParameterSet, Mandatory = true, 
+            ValueFromPipelineByPropertyName = true, HelpMessage = "The deployment name.")]
+        [ValidateNotNullOrEmpty]
+        public string DeploymentName { get; set; }
+
+        [Parameter(ParameterSetName = SaveAzureDeploymentTemplateCmdlet.DeploymentObjectParameterSet, Mandatory = true,
+            ValueFromPipeline = true, HelpMessage = "The deployment object.")]
+        public PSDeployment DeploymentObject { get; set; }
+
+        /// <summary>
+        /// Gets or sets the file path.
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The output path of the template file.")]
+        [ValidateNotNullOrEmpty]
+        public string Path { get; set; }
+
+        /// <summary>
+        /// Gets or sets the force parameter.
+        /// </summary>
+        [Parameter(Mandatory = false, HelpMessage = "Do not ask for confirmation.")]
+        public SwitchParameter Force { get; set; }
+
+        /// <summary>
+        /// Executes the cmdlet.
+        /// </summary>
+        protected override void OnProcessRecord()
+        {
+            base.OnProcessRecord();
+
+            var deploymentName = !string.IsNullOrEmpty(this.DeploymentName) ? this.DeploymentName : this.DeploymentObject.DeploymentName;
+
+            if (ShouldProcess(deploymentName, VerbsData.Save))
+            {
+                var template = ResourceManagerSdkClient.GetDeploymentTemplate(deploymentName);
+
+                string path = FileUtility.SaveTemplateFile(
+                    templateName: deploymentName,
+                    contents: template,
+                    outputPath:
+                        string.IsNullOrEmpty(this.Path)
+                            ? System.IO.Path.Combine(CurrentPath(), deploymentName)
+                            : this.TryResolvePath(this.Path),
+                    overwrite: this.Force,
```

Comment:
@Tiano2017 just to confirm, this call to `SaveTemplateFile` will prompt the user if the file they are trying to write to exists and `-Force` is not provided, correct?

Reply:
that is correct.

Reply:
@Tiano2017 got it, thanks for clarifying

Reply:
Yes. 

---
In reply to: [205823842](https://github.com/Azure/azure-powershell/pull/6516#discussion_r205823842) [](ancestors = 205823842)